kratos/mpi/tests/sources/test_pointer_communicator.cpp

```diff
@@ -186,9 +187,67 @@ KRATOS_TEST_CASE_IN_SUITE(PointerCommunicatorConstructByFunctor, KratosMPICoreFa
     }
 
 
-};
+}
 
+KRATOS_TEST_CASE_IN_SUITE(PointerMapCommunicator, KratosMPICoreFastSuite)
+{
+    DataCommunicator& r_default_comm = ParallelEnvironment::GetDefaultDataCommunicator();
+    Model current_model;
+    auto& mp = current_model.CreateModelPart("mp");
+    mp.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    const int world_size = r_default_comm.Size();
+    const int current_rank = r_default_comm.Rank();
 
+    auto pnode = mp.CreateNewNode(current_rank+1, current_rank,current_rank,current_rank); //the node is equal to the current rank;
+    pnode->FastGetSolutionStepValue(PARTITION_INDEX) = current_rank;
+    pnode->SetValue(TEMPERATURE, current_rank);
+
+    //we will gather on every node the global pointers of the nodes with index from
+    //current_rank(+1) to world_size
+    std::vector<int> indices;
+    for(int i=1; i<=world_size; ++i) {
+        indices.push_back(i);
+    }
+
+    auto gp_list = GlobalPointerUtilities::RetrieveGlobalIndexedPointers(mp.Nodes(), indices, r_default_comm);
+
+    // creates the gp map
+    GlobalPointerMapCommunicator<Node<3>, double> pointer_map_comm(r_default_comm, gp_list, 0.0);
+
+    // creates apply proxy
+    auto apply_temperature_assemble_proxy = pointer_map_comm.GetApplyProxy(
+        // local gp value update method
+        [](Node<3>& rNode, const double& NewValue) {
```

Comment:
why do you need two distinct functors, one for local and one for non local? i think you could do with the same

Reply:
I am not sure how we can achieve this in the following situation:
I was thinking for some reason, if user wants not the summation, may be multiplication like the following
`rCurrentValue *= rNewValue`; (or some weird computation using data from the captures as well).
In this case, he needs two different methods, one for updating the actual entity, one for updating the non_local_gp_map because non_local_gp_map holds pointers to entities which does not exist in their processes. So, their entities cannot be passed to the same method.

Reply:
the cool thing of the proxy, is that the function will be executed by the cpu that owns the pointer, so i don't think you are right about your concern.

Reply:
done  as discussed :)